ReactiveHeimdall/ReactiveHeimdall.swift

```diff
@@ -4,56 +4,133 @@ import ReactiveCocoa
 
 extension Heimdall {
     
-    /// Requests an access token with the resource owner's password credentials.
-    ///
-    /// :param: username The resource owner's username.
-    /// :param: password The resource owner's password.
-    ///
-    /// :returns: A signal that sends a `RACUnit` and completes when the
-    ///     request finishes successfully or sends an error of the request
-    ///     finishes with an error.
-    @objc
-    public func requestAccessToken(username: String, password: String) -> RACSignal {
-        return RACSignal.createSignal { subscriber in
+    /**
+        Requests an access token with the resource owner's password credentials.
+    
+        :param: username The resource owner's username.
+        :param: password The resource owner's password.
+    
+        :returns: A SignalProducer that completes when the request finishes
+            successfully or sends an error of the request finishes with an error.
```

Comment:
Well, if we want to be super correct, the `SignalProducer` never completes. Instead, this method returns `A SignalProducer that, when started, creates a signal that completes when ...`